check for minimum  wayland-client version (>= 1.20)

smohantty opened the pull request:
build fails if the installed wayland-client version is less than 1.20

smohantty committed:
check for minimum  wayland-client version (>= 1.20)

build fails if the installed wayland-client version is less than 1.20

deralmas commented on platform/linuxbsd/detect.py:
```diff
@@ -76,6 +77,21 @@ def check_wayland_dependencies():
+
+    import subprocess
+
+    wayland_client_version = subprocess.check_output(["pkg-config", "wayland-client", "--modversion"]).decode("utf-8")
```
Oh, thanks for caring to such a detail with such a (for now) broken project. Is `decode("utf-8")` needed? Bullet version handling does the same thing, but only using `strip()`. Did you have any issue with it?

smohantty replied:
My dev env is ubuntu 20.04 which has wayland-client 1.18, so thought of adding the patch before installing 1.20.
Yes I copied from bullet and stripe() was not working not sure why though. so used decode() version as it is being used in other parts of godot.

Regarding caring for the broken project, Actually, I work for a company and my task for this year is to support Wayland in Godot :) . I was planning to develop it internally and use it then I came across your patch. So thought of contributing here rather than duplicating the effort without any plan to upstream it.

I know you work on this as a side project so not sure about your plan for this patch. If you could share your thoughts that will be really helpful for me.

Thanks and have a great weekend.

deralmas replied:
@smohantty Hi! sorry for the delay. I just now found some time to answer to your question.

I still appreciate your desire to avoid duplication of work! Regarding my plans for the Wayland implementation, while it is indeed a side project that I do for fun and necessity (I use Wayland myself on my laptop and I don't want to install XWayland), I aim to make it as complete as possible, although I don't have a touchscreen. I don't have a fixed schedule and I'm pretty busy during the week, so it may not be fully polished in time for Godot 4.0. I think that may be something to keep in mind.

smohantty committed:
fix build break
CursorShape is defined as a macro in x11 header files which causing name conflict
with the CursorShape defination in DisplayServer.

Just followed display_server_x11 header file for the fix to undef CursorShape.

Riteo commented:
Thanks! Also, btw, I updated the main PR's OP to contain a general idea of my plans with the implementation.
